addons/html_editor/static/tests/utils/dom_info.test.js

```diff
@@ -359,3 +363,75 @@ describe("getDeepestPosition", () => {
         expect([node, offset]).toEqual([zwnbsp, 0]);
     });
 });
+
+describe("isemptyBlock", () => {
+    test("should identify span with icon classes as non-empty", () => {
+        const [span] = insertTestHtml('<span class="fa fa-trash-o"></span>');
+        const result = isEmptyBlock(span);
+        expect(result).toBe(false);
+    });
+
+    test("should identify p with span containing icon classes as non-empty", () => {
+        const [p] = insertTestHtml('<p><span class="fa fa-trash-o"></span></p>');
+        const result1 = isEmptyBlock(p);
+        const result2 = isEmptyBlock(p.firstChild);
+        expect(result1).toBe(false);
+        expect(result2).toBe(false);
+    });
+
+    test("should identify p with single br tag as empty and br tag as non-empty", () => {
+        const [p] = insertTestHtml("<p><br></p>");
```

Comment:
You can add one more p tag with 2 br element here
```suggestion
        const [p1, p2] = insertTestHtml("<p><br></p><p><br><br></p>");
```
and test `isEmptyBlock` for both.